# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Sweeping.cpp
# Commit: 28829816a629
# Full Hash: 28829816a629b28125db3b4c65c5e22518e167e8
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-09 21:37:17
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 6: Stop using JSContext for GC helper thread tasks r=jandem
#   
#   This patch moves the debug-only GC state from JSContext to JSFreeOp. Support
#   functions for this move to FreeOp.h and most RAII classes move to GCInternals.h.
#   
# ==============================================================================

diff -r 410d1b57bb09 -r 28829816a629 js/src/gc/Sweeping.cpp
--- a/js/src/gc/Sweeping.cpp	Wed Mar 09 10:17:07 2022 +0000
+++ b/js/src/gc/Sweeping.cpp	Wed Mar 09 10:17:08 2022 +0000
@@ -319,7 +319,7 @@
     return;
   }
 
-  JSFreeOp* fop = TlsContext.get()->defaultFreeOp();
+  JSFreeOp* fop = TlsFreeOp.get();
 
   // Sweep zones in order. The atoms zone must be finalized last as other
   // zones may have direct pointers into it.
@@ -456,7 +456,7 @@
 
     lifoBlocks.freeAll();
 
-    JSFreeOp* fop = TlsContext.get()->defaultFreeOp();
+    JSFreeOp* fop = TlsFreeOp.get();
     for (Nursery::BufferSet::Range r = buffers.all(); !r.empty();
          r.popFront()) {
       // Malloc memory associated with nursery objects is not tracked as these